object/ldap_conn.go

```diff
@@ -395,32 +416,30 @@ func SyncLdapUsers(owner string, syncUsers []LdapUser, ldapId string) (existUser
 				newUser.SignupApplication = organization.DefaultApplication
 			}
 
-			affected, err := AddUser(newUser)
+			affected, err = AddUser(newUser)
 			if err != nil {
-				return nil, nil, err
-			}
-
-			if !affected {
-				failedUsers = append(failedUsers, syncUser)
-				continue
+				return nil, err
 			}
 		}
+		if !affected {
+			failedUsers = append(failedUsers, syncUser)
+			continue
+		}
 
 		ldap, err := GetLdap(ldapId)
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
 
 		if ldap.EnableRoleMapping {
 			err = SyncRoles(syncUser, name, owner)
```

Comment:
Should we sync roles for existed users?

Reply:
why not? Can't there be the situation when any mapping was changed and user can get new role according to it